### ISSUE TITLE
DashboardNotification: Survey old version will show up

### ISSUE THREAD

**Issue by paulbert** [open-learning-exchange/planet#5928]
Steps to reproduce:

1. Create a survey
2. Send it to a user
3. Edit that survey
4. Send it to the same user
5. Log in as that user
6. Ignore the dialog, and only complete the surveys listed in **mySurveys**
7. Logout/login
8. The dialog will popup again with the survey, but it won't show up in **mySurveys**

After a survey is edited and resent, we no longer list that survey in **mySurveys**.  This is correct, the old survey should be ignored if it was "pending".

Labeling this **low priority** since after #5091 is completed this specific instance will no longer be an issue, but we should work to make sure the list in the dialog matches the list in **mySurveys** in all situations.
